test/eventlistener_test.go

```diff
@@ -35,13 +37,26 @@ func TestEventListenerCreate(t *testing.T) {
 
 	t.Log("Start EventListener e2e test")
 
+	// Create EventListener SA
+	sa, err := c.KubeClient.CoreV1().ServiceAccounts(namespace).Create(
+		&corev1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "default-sa",
+			},
+		})
+	if err != nil {
+		t.Fatalf("Failed to create ServiceAccount: %s", err)
+	}
+	t.Logf("Created ServiceAccount %s in namespace %s", sa.Name, sa.Namespace)
+
 	// Create EventListener
 	el, err := c.TriggersClient.TektonV1alpha1().EventListeners(namespace).Create(
 		&v1alpha1.EventListener{
 			ObjectMeta: metav1.ObjectMeta{
 				Name: "my-eventlistener",
 			},
 			Spec: v1alpha1.EventListenerSpec{
+				ServiceAccountName: "my-serviceaccount",
```

Comment:
I think the name you created above was `"default-sa"` instead of `"my-serviceaccount"`. Maybe just use `sa.Name`? 👍